Tribler/Core/Upgrade/db72_to_pony.py

```diff
@@ -385,22 +394,14 @@ def mark_conversion_finished(self):
             else:
                 self.mds.MiscData(name=CONVERSION_FROM_72, value=CONVERSION_FINISHED)
 
+    @inlineCallbacks
     def do_migration(self):
         self.convert_personal_channel()
-
-        def background_stuff():
-            self.mds.clock = None  # We should never touch the clock during legacy conversions
-            if not self.shutting_down:
-                self.convert_discovered_torrents()
-            if not self.shutting_down:
-                self.convert_discovered_channels()
-            if not self.shutting_down:
-                self.update_trackers_info()
-            if not self.shutting_down:
-                self.mark_conversion_finished()
-            self.mds._db.disconnect()
-
-        return deferToThread(background_stuff)
+        self.mds.clock = None  # We should never touch the clock during legacy conversions
```

Comment:
Are we sure this is safe to do? There might be a better (potentially safer) solution to prevent the clock from being touched, e.g. by setting a flag in the `mds` object.

Reply:
Actually, this *is* the safest solution. The thing is here mostly for the programmer's convenience. Upgrading is and should be a completely deterministic process. We *want* Tribler to crash if the clock is touched during the conversion process, because *this should never, ever happen*. Any other behavior can potentially create nearly-uncatchable inconsistencies in produced metadata entities.

Reply:
I think that the Python exception system is built for detection of this kind of behaviour. If we adopt a defensive programming style, we should have some kind of flag whether it can be touched and if it is touched when this flag is set, we raise a `RuntimeError`. However, this might be a bit too much overengineering for this specific situation so I guess it's fine like this.